apps/src/gamelab/PropTypes.js

```diff
@@ -0,0 +1,149 @@
+/** @file Type definitions (React and otherwise) specific to Gamelab */
+
+import _ from 'lodash';
+import React from 'react';
+
+/**
+ * @typedef {Object} Vector2
+ * @property {number} x
+ * @property {number} y
+ */
+const Vector2 = React.PropTypes.shape({
+  x: React.PropTypes.number.isRequired,
+  y: React.PropTypes.number.isRequired
+});
+
+/*
+ * MIGRATION NOTES - 30 June 2016 bbuchanan
+ *  The old shape of animations is an array of animation metadata.
+ *    [
+ *      {animation},
+ *      {animation}
+ *    ]
+ *
+ *  The new shape is an object with animation and cache components
+ *    {
+ *     orderedKeys: [ AnimationKey, AnimationKey ],
+ *     propsByKey: {
+ *       AnimationKey: {AnimationProps},
+ *       AnimationKey: {AnimationProps}
+ *     }
+ *   }
+ *
+ *  AnimationProps should include the actual spritesheet (as blob and dataURI),
+ *  dimensions and frame dimensions, framerate, name, last save time, version IDs,
+ *  URL to fetch the animation from the API, etc.
+ *
+ *  We serialize a smaller set of information {SerializedAnimationProps}.
+ *
+ *  We need to do a migration if the old style gets loaded.
```

Comment:
Haha - that migration [exists here](https://github.com/code-dot-org/code-dot-org/pull/9259/files#diff-17f6f980ad057a6db9a3b1f7de1dcd41R226).  I'll update this comment.